internal/wiz/structs.go

```diff
@@ -244,6 +244,17 @@ type SAMLGroupMappingCreateInput struct {
 	Projects        []string `json:"projects"`
 }
 
+// UpdateSAMLGroupMappingInput struct
+type UpdateSAMLGroupMappingInput struct {
+	ID    string                      `json:"id"`
+	Patch ModifySAMLGroupMappingPatch `json:"patch"`
+}
+
+// ModifySAMLGroupMappingPatch struct
```

Comment:
The type should be defined and implemented as per the graphql schema, including a slice of string for `delete` (with `omitempty` as required). Then there should also be no need for the Delete custom types defined in the resource. 

`upsert` is also of type `SAMLGroupDetailsInput`, not `SAMLGroupMappingUpdateInput` so a new struct definition is needed for this (the definition may be duplicated but it will then mirror the vendor schema). You'll see that the `description` field is not yet added as it is part of a later evolution of the schema, you can choose to either add it or add a struct comment that it is incomplete (there are examples of this in the structs file). 

<img width="408" alt="image" src="https://github.com/user-attachments/assets/6554b8a1-ab8e-4032-a250-d135bb99feac">

Getting the graphql schema modelling was historically a bit of a challenge; for the structs file, when there is a resource introduced, it should be respected as far as possible. Custom types should then also go in the provider package and not in the wiz/structs file.

For future reference, there are a few notes in the top of the structs.go file with a few guidelines.

Reply:
Ah yeah sorry, now I know what you meant. I changed it and it should be fine now I think.

Reply:
Must be:

```golang
type ModifySAMLGroupMappingPatch struct {
	Upsert * SAMLGroupDetailsInput `json:"upsert,omitempty"`
	Delete *[]string                     `json:"delete,omitempty"`
}
```